crates/router/src/connector/tsys/transformers.rs

```diff
@@ -37,9 +67,11 @@ pub struct TsysPaymentAuthSaleRequest {
     order_number: String,
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for TsysPaymentsRequest {
+impl TryFrom<&TsysRouterData<&types::PaymentsAuthorizeRouterData>> for TsysPaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+    fn try_from(
+        item: &TsysRouterData<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
```

Comment:
Instances where `item.request` is utilized should be substituted with `item.router_data.request`

In case you have any queries, you can ask them on this PR thread, or on [slack](https://join.slack.com/t/hyperswitch-io/shared_invite/zt-1k6cz4lee-SAJzhz6bjmpp4jZCDOtOIg) whichever you are comfortable with